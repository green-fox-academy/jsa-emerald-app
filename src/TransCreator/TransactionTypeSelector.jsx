import React from 'react';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './Style';
import setThemeStyle from '../Common/theme/setThemeStyle';

const TransactionTypeSelector = ({ transType, setTransType }) => {
  const { themeMode } = useSelector((state) => state.theme);
  const theme = setThemeStyle(themeMode);
  return (
    <View style={styles.typeButtonGroup}>
      <Button
        buttonStyle={transType === 'Expense'
          ? [styles.typeButton, styles.typeButtonActive, { borderColor: theme.mainColor.color }] : styles.typeButton}
        titleStyle={transType === 'Expense'
          ? [styles.typeTitle, styles.typeTitleActive, theme.mainColor] : styles.typeTitle}
        title="Expense"
        type="outline"
        onPress={() => setTransType('Expense')}
      />
      <Button
        buttonStyle={transType === 'Income'
          ? [styles.typeButton, styles.typeButtonActive, { borderColor: theme.mainColor.color }] : styles.typeButton}
        titleStyle={transType === 'Income'
          ? [styles.typeTitle, styles.typeTitleActive, theme.mainColor] : styles.typeTitle}
        title="Income"
        type="outline"
        onPress={() => setTransType('Income')}
      />
    </View>
  )
};

TransactionTypeSelector.propTypes = {
  transType: PropTypes.string,
  setTransType: PropTypes.func,
};

TransactionTypeSelector.defaultProps = {
  transType: 'Expense',
  setTransType: null,
};

export default TransactionTypeSelector;
