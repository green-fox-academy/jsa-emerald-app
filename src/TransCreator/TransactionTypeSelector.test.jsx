import React from 'react';
import renderer from 'react-test-renderer';
import TransactionTypeSelector from './TransactionTypeSelector';

describe('<TransactionTypeSelector/>', () => {
  it('render Expense', () => {
    const tree = renderer.create(<TransactionTypeSelector
      transType="Expense"
      setTransType={jest.fn()}
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render Income', () => {
    const tree = renderer.create(<TransactionTypeSelector
      transType="Income"
      setTransType={jest.fn()}
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
