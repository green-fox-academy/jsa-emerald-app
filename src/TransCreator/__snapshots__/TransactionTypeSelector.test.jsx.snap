// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionTypeSelector/> render Expense 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "black",
            "borderRadius": 50,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 15,
            "padding": 8,
            "paddingBottom": 5,
            "paddingTop": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Expense
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 50,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "gray",
            "borderRadius": 50,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 15,
            "padding": 8,
            "paddingBottom": 5,
            "paddingTop": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 15,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Income
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<TransactionTypeSelector/> render Income 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 50,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "gray",
            "borderRadius": 50,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 15,
            "padding": 8,
            "paddingBottom": 5,
            "paddingTop": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 15,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Expense
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 3,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "black",
            "borderRadius": 50,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 0,
            "marginLeft": 6,
            "marginRight": 6,
            "marginTop": 15,
            "padding": 8,
            "paddingBottom": 5,
            "paddingTop": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Income
        </Text>
      </View>
    </View>
  </View>
</View>
`;
